src/core/probe/kernel/mod.rs

```diff
@@ -109,6 +134,18 @@ trait ProbeBuilder {
     fn attach(&mut self, target: &str) -> Result<()>;
 }
 
+fn reuse_map_fds(open_obj: &libbpf_rs::OpenObject, map_fds: &Vec<(String, i32)>) -> Result<()> {
+    if !map_fds.is_empty() {
+        for map in map_fds {
```

Comment:
another nit based on locality :D 
What about using explicit .iter() here?
I may be wrong, but I find implicit conversion a bit error prone as it may vary depending on the fact the collection is a [mut] ref or not